### PR TITLE
fix color multiply error

### PR DESCRIPTION
**Pull request by minggo**
### Changelog

* two colors multiply should divide by 255

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
@cocos-robot run test cases

**Comment by minggo** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

File: cocos/core/math/color.ts [hunk 1]
```diff
@@ -734,7 +734,11 @@ export class Color extends ValueType implements Modifiable {
      * @param other The specified color.
      */
     public multiply (other: Color): Color {
-        Color.multiply(this, this, other);
+        // FIXME: not sure if other is really Color, so use getter.
+        this._data[R_INDEX] = this._data[R_INDEX] * other.r / 255;
```
Comment (dumganhar) [inline cocos/core/math/color.ts]: Could use ` this._data[R_INDEX] *= other.r / 255` to save some bytes.

**Comment by sushanhong** [comment]
@cocos-robot run test cases

File: cocos/core/math/color.ts (hunk 1 above)
Comment (minggo) [reply cocos/core/math/color.ts]: You are right. 
Fixed.

**Pull request merged**